### PR TITLE
Use unsigned comparison operators for unsigned SIMD types.

### LINKED ISSUE #21719: SIMD comparison operators are always signed.
The compiler always emits signed SIMD comparison instructions, even for unsigned element types:
```rust
use std::simd::u32x4;
u32x4(1, 2, 3, 1 << 31) < u32x4(2, 3, 4, 0)
// => u32x4(4294967295, 4294967295, 4294967295, 4294967295)
// i.e. <true, true, true, true>. expected <true, true, true, false>
```

### PR DESCRIPTION
**Pull request by huonw**
Previously comparisons of SIMD types were always signed, even unsigned
comparisons, meaning 0xFFFF_FFFF_u32 < 0 inside a SIMD vector.

Fixes #21719.

### CODE REVIEW

File: src/librustc_trans/trans/base.rs [hunk 1]
```diff
@@ -623,7 +623,7 @@ pub fn compare_simd_types<'blk, 'tcx>(
                     size: uint,
                     op: ast::BinOp)
                     -> ValueRef {
-    match t.sty {
+    let cmp = match t.sty {
```
Comment (eddyb) [inline src/librustc_trans/trans/base.rs]: Nice! I did something similar in my branch, though I had all the matching in another function and only needed the signedness from this one.

File: src/test/run-pass/simd-binop.rs [hunk 2]
```diff
@@ -55,17 +55,18 @@ pub fn main() {
...
-    assert!(eq_i32x4(i32x4(1, 2, 3, 4) == i32x4(3, 2, 1, 0), i32x4(0, !0, 0, 0)));
-    assert!(eq_i32x4(i32x4(1, 2, 3, 4) != i32x4(3, 2, 1, 0), i32x4(!0, 0, !0, !0)));
-    assert!(eq_i32x4(i32x4(1, 2, 3, 4) < i32x4(3, 2, 1, 0), i32x4(!0, 0, 0, 0)));
-    assert!(eq_i32x4(i32x4(1, 2, 3, 4) <= i32x4(3, 2, 1, 0), i32x4(!0, !0, 0, 0)));
-    assert!(eq_i32x4(i32x4(1, 2, 3, 4) >= i32x4(3, 2, 1, 0), i32x4(0, !0, !0, !0)));
-    assert!(eq_i32x4(i32x4(1, 2, 3, 4) > i32x4(3, 2, 1, 0), i32x4(0, 0, !0, !0)));
+    assert!(eq_i32x4(i32x4(1, 2, 3, -1) == i32x4(3, 2, 1, 0), i32x4(0, !0, 0, 0)));
+    assert!(eq_i32x4(i32x4(1, 2, 3, -1) != i32x4(3, 2, 1, 0), i32x4(!0, 0, !0, !0)));
+    assert!(eq_i32x4(i32x4(1, 2, 3, -1) < i32x4(3, 2, 1, 0), i32x4(!0, 0, 0, !0)));
+    assert!(eq_i32x4(i32x4(1, 2, 3, -1) <= i32x4(3, 2, 1, 0), i32x4(!0, !0, 0, !00)));
```
Comment (tbu-) [inline src/test/run-pass/simd-binop.rs]: `!00`?

Comment (huonw) [inline src/test/run-pass/simd-binop.rs]: Just a typo, has no semantic impact (I'll fix it though).

**Pull request merged**